.github/workflows/build.yml

```diff
@@ -38,12 +38,13 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-go@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
         with:
           go-version: '1.22'
           check-latest: true
-      - uses: actions/checkout@v4
+          cache-dependency-path: go.mod
       - name: Lint
-        run: |
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.57.2
-          ./bin/golangci-lint run
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.57.2
```

Comment:
See https://golangci-lint.run/welcome/install/#github-actions. The `golangci-lint` GitHub Action has built-in cache